provider/execution/execution.go

```diff
@@ -95,20 +100,30 @@ func (e TableExecutor) doMultiplexResolve(ctx context.Context, clients []schema.
 		diagsChan      = make(chan diag.Diagnostics)
 		totalResources uint64
 	)
+	var (
+		allDiags    diag.Diagnostics
+		doneClients = 0
+	)
 	logger := clients[0].Logger()
 	logger.Debug("multiplexing client", "count", len(clients), "table", e.Table.Name)
 	defer close(diagsChan)
 	for _, client := range clients {
+		// we can only limit on a granularity of a top table otherwise we can get deadlock
+		if parent == nil {
```

Comment:
no need to check parent != nil, as table relation resolvers don't get multiplexed.